### PR TITLE
Updates and additions to the composite R2R format

### PR DESCRIPTION
**Pull request by trylek**
Based on my progress with composite R2R implementation I propose
clarifying the design as follows:

1) I have proposed a rewriting scheme for the standalone MSIL
files next to the composite R2R executable to receive forwarding
information pointing at the composite file;

2) I have updated the section names I renamed yesterday and I
added a description of the new section OwnerCompositeExecutable
to be used for the forwarding information.

Thanks

Tomas

### CODE REVIEW

File: docs/design/coreclr/botr/readytorun-format.md [hunk 1]
```diff
@@ -507,7 +513,7 @@ Foreign RIDs are only present if a fragile inlining was allowed at compile time.
 **TODO:** It remains to be seen whether `READYTORUN_SECTION_METHODCALL_THUNKS` and / or
```
Comment (jkotas) [inline docs/design/coreclr/botr/readytorun-format.md]: `READYTORUN_SECTION_INLINING_INFO` should be renamed (multiple places in the document).

File: docs/design/coreclr/botr/readytorun-format.md [hunk 2]
```diff
@@ -136,33 +141,34 @@ and execute the ready to run file.
...
+
+## CompilerIdentifier
 
 This section contains zero terminated ASCII string that identifies the compiler used to produce the
 image.
 
 **Example**: `CoreCLR 4.6.22727.0 PROJECTK`
 
-## READYTORUN_SECTION_IMPORT_SECTIONS
+## ImportSections
```
Comment (jkotas) [inline docs/design/coreclr/botr/readytorun-format.md]: Nit: I may be nice to include the fully qualified names in these chapters, e.g. `ReadyToRunSectionType.ImportSections`

Comment (trylek) [reply docs/design/coreclr/botr/readytorun-format.md]: Good idea, will fix.

**Review by jkotas — APPROVED** [review]

File: docs/design/coreclr/botr/readytorun-format.md (hunk 1 above)
Comment (trylek) [reply docs/design/coreclr/botr/readytorun-format.md]: Fixed, thanks for spotting!

File: docs/design/coreclr/botr/readytorun-format.md (hunk 2 above)
Comment (trylek) [reply docs/design/coreclr/botr/readytorun-format.md]: Fixed in 2nd commit.

**Pull request merged**